src/include/add-on/add-on-workflow.rb

```diff
@@ -1175,32 +1121,29 @@ def RunAddOnMainDialog(enable_back, enable_next, enable_abort, back_button, next
 
           if ret2 == :next
             # Add-On product has been added, integrate it (change workflow, use y2update)
-            AddOnProduct.Integrate(AddOnProduct.src_id)
+            @added_repos.each { |src_id| AddOnProduct.Integrate(src_id) }
 
             # check whether it requests registration (FATE #301312)
-            AddOnProduct.PrepareForRegistration(AddOnProduct.src_id)
+            @added_repos.each { |src_id| AddOnProduct.PrepareForRegistration(src_id) }
```

Comment:
can it be done in one step?
like
```
@added_repos.each do |src_id|
  AddOnProduct.Integrate(src_id)
  AddOnProduct.PrepareForRegistration(src_id)
end
```

Reply:
I'm not sure, the methods call quite a lot of other functions, I just added a comment to possibly fix that later.